RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -85,6 +85,12 @@ public final class PaywallViewController: UIViewController {
         self.hostingController.view.frame = self.view.bounds
     }
 
+    public override func viewDidDisappear(_ animated: Bool) {
+        if isBeingDismissed {
+            self.delegate?.paywallViewControllerDismissed?(self)
```

Comment:
I'm not certain this is the best way to do this. Afaik, in iOS, the presenter is usually in charge of dismissing the presented VC, but not sure if the same applies to modals, where it can be dismissed by user interaction. And we want to know when it's dismissed so the hybrids can return a result at that point in time. Any feedback would be welcome @aboedo @NachoSoto

Reply:
I guess in the hybrids, it's probably not a modal though, so this might not work...

Reply:
the footer wouldn't own the modal anyway, right? I think we should just leave it up to the owner of the view that has a footer in it to dismiss the whole thing, which in turn will kill the footer

Reply:
Right, this is a bit confusing, but this fix is for the old approach of presenting the paywall as a modal in the hybrids and fix the issue I mentioned in the description... But yeah, if we end up moving to using views, this becomes unnecessary.